src/network/networkd-link.c

```diff
@@ -2481,10 +2449,14 @@ static int link_configure(Link *link) {
         if (r < 0)
                return r;
 
-        r = link_set_ipv6_proxy_ndp(link);
+        r = ipv6_proxy_ndp_set(link);
         if (r < 0)
                return r;
 
+        r = ipv6_proxy_ndp_addresses_configure(link);
```

Comment:
Maybe `networkd` could automatically enable `proxy_ndp` once an address is configured.